apps/dcos_l4lb/src/dcos_l4lb_mgr.erl

```diff
@@ -176,7 +200,8 @@ push_routes(Namespace, #state{routes = Routes, route_mgr = RouteMgr}) ->
 push_services(Namespace, State = #state{last_configured_vips = VIPs}) ->
     lists:foreach(fun({VIP, BEs}) -> add_service(VIP, BEs, Namespace, State) end, VIPs).
 
-do_reconcile(VIPs0, State0 = #state{ns = Namespaces}) ->
+do_reconcile(VIPs, State0 = #state{ns = Namespaces}) ->
+    VIPs0 = vips_port_mappings(VIPs),
     VIPs1 = process_reachability(VIPs0, State0),
     Routes = ordsets:from_list([VIP || {{_Proto, VIP, _Port}, _Backends} <- VIPs1]),
     State1 = State0#state{last_received_vips = VIPs0, last_configured_vips = VIPs1, routes = Routes},
```

Comment:
VIPs instead of VIPs0

Reply:
we must store original value into the state for future reconciliation, because port mappings can be changed, but lashup wouldn't get such changes.

Reply:
yeah, this make sense